api/handlers.go

```diff
@@ -1302,3 +1338,27 @@ func userLimitsGetFromTier(tierID int, quotaExceeded, inBytes bool) *UserLimitsG
 		RegistryDelay:     limitsTier.RegistryDelay,
 	}
 }
+
+// validateIP is a simple pass-through helper that returns valid IPs as they are
+// and returns an empty string for invalid IPs.
+func validateIP(ip string) string {
```

Comment:
I didn't know it was there and none of the SO articles I read about parsing IPs with go mentioned it.

I'll switch to it immediately, it will make this PR so much nicer. Thanks!